ddl/ddl_api.go

```diff
@@ -347,6 +359,26 @@ func columnDefToCol(ctx sessionctx.Context, offset int, colDef *ast.ColumnDef) (
 				// TODO: Support this type.
 			}
 		}
+
+		// primary key should not be null
+		if mysql.HasPriKeyFlag(col.Flag) && hasDefaultValue && col.DefaultValue == nil {
+			return nil, nil, types.ErrInvalidDefault.GenByArgs(col.Name)
+		}
+		// set primary key flag for outer primary key constraint.
+		// such as: create table t1 (id int , age int, primary key(id))
+		if !mysql.HasPriKeyFlag(col.Flag) && outPriKeyConstraint != nil && outPriKeyConstraint.Tp == ast.ConstraintPrimaryKey {
+			for _, key := range outPriKeyConstraint.Keys {
+				if key.Column.Name.L != col.Name.L {
+					continue
+				}
+				col.Flag |= mysql.PriKeyFlag
```

Comment:
we do not set the PriKeyFlag in this case before?

Reply:
For outer primary key constraint, do not set before.